src/k8s/pkg/k8sd/features/features.go

```diff
@@ -0,0 +1,11 @@
+package features
+
+const (
+	DNS           string = "dns"
```

Comment:
let's add a `type FeatureName string` and make these `DNS FeatureName = "dns"` etc

Then, in all functions that require the name of a feature, we use `FeatureName` as the type instead. This is not as good as an enum, but it's better to force us to think about not passing in arbitrary values.

Reply:
good point, thanks. in that case, is it ok if we use `string(features.Network)` in the places that we expect strings? 
places like this:
```go
    for _, component := range strings.FieldsFunc(components, func(r rune) bool { return unicode.IsSpace(r) || r == ',' }) {
		switch component {
		case string(features.Network):
			config.ClusterConfig.Network.Enabled = utils.Pointer(true)
		...
		}
	}
```